server/entity/MediaRequest.ts

```diff
@@ -242,6 +242,22 @@ export class MediaRequest {
           }, [] as number[]);
       }
 
+      // We should also check seasons that are available/partially available but don't have existing requests
+      if (media.seasons) {
+        existingSeasons = [
+          ...existingSeasons,
+          ...media.seasons
+            .filter(
+              (season) =>
+                season[requestBody.is4k ? 'status4k' : 'status'] ===
+                  MediaStatus.AVAILABLE ||
+                season[requestBody.is4k ? 'status4k' : 'status'] ===
+                  MediaStatus.PARTIALLY_AVAILABLE
```

Comment:
I think this still needs a filter for seasons that are have been marked as "Requested", I'm was still experiencing the same issue but it seems this is the reason why 😃

Reply:
Also, I think theres still an issue with whatever logic is used to trigger Sonarr to search for Seasons. Season 11 is the only one missing from AHS as it hasn't aired yet, and we know the only reason AHS is getting added to Overseerr requests (right now) is because it's missing another filter?. BUT, what's curious is that it's still attempting to re-download every single season** 🤔 

**Overseerr only has the badge for S11 under requests

Reply:
I'm confused. Is it requesting things that are already requested?

Reply:
When words fail, basically this video will show you what I described above ^

https://vimeo.com/740224465

Side-note, while recording this I came across another bug where it added the entry twice because it appeared on two watchlists 👀

Reply:
I see. Because you have it present in Sonarr/Radarr. I will fix it.

As for double requests, I can fix this by running each users watchlist requests in succession instead of asynchronously.

Reply:
Hopefully I won't encounter anymore issues after this, feel like I'm teasing you every time you get close to a PR haha

On that note (sorry again), but I just noticed that if you modify a request that has synced from the watchlist and change it to request X amount of seasons instead of all of them, the next sync will add another request for the missing seasons 😅

Reply:
Pushed a fix for the other issues. Try it out.

> On that note (sorry again), but I just noticed that if you modify a request that has synced from the watchlist and change it to request X amount of seasons instead of all of them, the next sync will add another request for the missing seasons 😅

That's not really a bug. Watchlist items for shows will always request for all seasons each time. We can't do granular requests here. If two users both add the same show, and you edit one? How do you know the other didn't want all seasons? How do we determine what to request for each user? I think it's either all or nothing. I imagine in most setups, admins will want to disable automated show requests :)

Reply:
No problem, I'm just waiting on the CI to publish and I'll let you know if I hit anymore issues.

> That's not really a bug.

Oh I assumed as much, people that do have it enabled will most likely have auto approve on too. It was a super small edge case I figured I may as well mention. Out of curiosity, with this fix, if I modified the request to include less seasons but marked the seasons I didn't want as monitored in Sonarr, it should skip the request on the next sync right?

Reply:
In its current state, it would make a new request for the missing seasons. it's all or nothing.

We are talking now about maybe adding a way to track automated season requests. So if you decline them, you don't get more from the same user.

Reply:
I actually didn't think of that, what would currently happen if a request is declined? Will it be stuck in an endless loop of requesting until it's accepted? If thats is the case, what would the best solution? Maybe some form of timed item blacklist per user?

Reply:
Im adding the ability to track auto requests. Each user will only get one auto-request per-media.

Reply:
Gotcha, assuming this data will be cached, will you be adding an option to flush it?

Reply:
Deleting the auto-request would allow new ones.